tests/components/wled/test_light.py

```diff
@@ -246,345 +242,336 @@ async def test_dynamically_handle_segments(
+    await hass.async_block_till_done()
 
-        state = hass.states.get("light.wled_rgb_light_segment_0")
-        assert state
-        assert state.attributes.get(ATTR_BRIGHTNESS) == 100
+    state = hass.states.get("light.wled_rgb_light")
+    assert state
+    assert state.attributes.get(ATTR_BRIGHTNESS) == 100
 
     # Test segment is off when master is off
     device.state.on = False
-    with patch(
-        "homeassistant.components.wled.coordinator.WLED.update",
-        return_value=device,
-    ):
-        async_fire_time_changed(hass, dt_util.utcnow() + SCAN_INTERVAL)
-        await hass.async_block_till_done()
-        state = hass.states.get("light.wled_rgb_light_segment_0")
-        assert state
-        assert state.state == STATE_OFF
+    async_fire_time_changed(hass, dt_util.utcnow() + SCAN_INTERVAL)
+    await hass.async_block_till_done()
+    state = hass.states.get("light.wled_rgb_light")
+    assert state
+    assert state.state == STATE_OFF
 
     # Test master is turned off when turning off a single segment
-    with patch("wled.WLED.master") as master_mock:
-        await hass.services.async_call(
-            LIGHT_DOMAIN,
-            SERVICE_TURN_OFF,
-            {ATTR_ENTITY_ID: "light.wled_rgb_light_segment_0", ATTR_TRANSITION: 5},
-            blocking=True,
-        )
-        await hass.async_block_till_done()
-        master_mock.assert_called_once_with(
-            on=False,
-            transition=50,
-        )
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "light.wled_rgb_light", ATTR_TRANSITION: 5},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    assert mock_wled.master.call_count == 1
+    mock_wled.master.assert_called_with(
+        on=False,
+        transition=50,
+    )
 
     # Test master is turned on when turning on a single segment, and segment
     # brightness is set to 255.
-    with patch("wled.WLED.master") as master_mock, patch(
-        "wled.WLED.segment"
-    ) as segment_mock:
-        await hass.services.async_call(
-            LIGHT_DOMAIN,
-            SERVICE_TURN_ON,
-            {
-                ATTR_ENTITY_ID: "light.wled_rgb_light_segment_0",
-                ATTR_TRANSITION: 5,
-                ATTR_BRIGHTNESS: 42,
-            },
-            blocking=True,
-        )
-        await hass.async_block_till_done()
-        master_mock.assert_called_once_with(on=True, transition=50, brightness=42)
-        segment_mock.assert_called_once_with(on=True, segment_id=0, brightness=255)
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_ENTITY_ID: "light.wled_rgb_light",
+            ATTR_TRANSITION: 5,
+            ATTR_BRIGHTNESS: 42,
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    mock_wled.segment.call_count == 1
+    mock_wled.master.call_count == 2
```

Comment:
Why do we set the call_count here?

Reply:
Missing asserts, I do actually have a local guard for that (which was not there because of a fresh venv). Should open a PR to add that to our CI :)